charm_test.go

```diff
@@ -219,3 +223,132 @@ func cloneDir(c *gc.C, path string) string {
 	c.Assert(err, gc.IsNil)
 	return newPath
 }
+
+// TestCreateMaybeCreateVersionFile verifies if the version file can be created
+// in case of git revision control directory
+func (s *CharmSuite) TestGitMaybeCreateVersionFile(c *gc.C) {
```

Comment:
given how much code is common in all these 3 functions, extract a help method called assertVersionFile(c *gc.C, dir, execName string, args ...string)
 Then the logic uses these params.
Each of the 3 test methods then become something like

func (s *CharmSuite) TestGitMaybeCreateVersionFile(c *gc.C) {
    s.assertVersionFile(c, ".git", "git", "describe", "--dirty")
}